File: go/chapter4/summing_a_million.go
package main

import "fmt"

func main() {
	numbers := []int{}

	for i := 1; i <= 1_000_000; i++ {
		numbers = append(numbers, i)
	}

	fmt.Println(min(numbers))
	fmt.Println(max(numbers))
	fmt.Println(sum(numbers))
}

func min(numbers []int) int {
	min := numbers[0]

	for _, n := range numbers {
		if n < min {
			min = n
		}
	}

	return min
}

func max(numbers []int) int {
	max := numbers[0]

	for _, n := range numbers {
		if n > max {
			max = n
		}
	}

	return max
}

func sum(numbers []int) int {
	total := 0

	for _, n := range numbers {
		total += n
	}

	return total
}

File: javascript/chapter4/summing_a_million.js
let numbers = [];

for (let i = 1; i <= 1_000_000; i++) {
	numbers.push(i);
}

console.log(min(numbers));
console.log(max(numbers));
console.log(sum(numbers));


/**
 * @param {Number[]} numbers
 * @returns {Number}
 */
function min(numbers) {
	let min = numbers[0];

	numbers.forEach(n => {
		if (n < min) min = n;
	});

	return min;
}

/**
 * @param {Number[]} numbers
 * @returns {Number}
 */
function max(numbers) {
	let max = numbers[0];

	numbers.forEach(n => {
		if (n > max) max = n;
	});

	return max;
}

/**
 * @param {Number[]} numbers
 * @returns {Number}
 */
function sum(numbers) {
	let total = 0;

	numbers.forEach(n => total += n);

	return total;
}
